.github/workflows/run_periodic_tests.yml

```diff
@@ -122,7 +122,17 @@ jobs:
         if: matrix.os == 'ubuntu-latest'
         run: python -m nox -s scripts
 
-  # M-series Mac Mini
+      - name: Cache vcpkg dependencies
+        if: matrix.os == 'windows-latest'
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/vcpkg
+          key: ${{ runner.os }}-vcpkg-${{ hashFiles('**/vcpkg.json') }}
+          restore-keys: |
+            ${{ runner.os }}-vcpkg-
+
+
```

Comment:
Besides this, we also need the PyBaMM-related environment variables, such as PYBAMM_USE_VCPKG_ON, CMAKE_BUILD_PARALLEL_LEVEL, and the rest present in `publish_pypi.py` so that `setup.py` can use all features of `vcpkg`.

After you add all of those, I would suggest triggering this workflow from this branch to test these changes (if you do it correctly, you should see a `win-amd64` file being built for PyBaMM rather than the current `none-any.whl`).

Reply:
@agriyakhetarpal How can I trigger a workflow from this branch?

Reply:
You'd have to go to the `actions` tab on your fork then select the workflow you want to run from the left panel. 
Click on the `Run Workflow` button >> Select the target branch >> Confirm after Clicking on the blue `Run Workflow` button


![image](https://github.com/pybamm-team/PyBaMM/assets/104268427/4640bc11-bb2e-482d-9427-54d0ae1d764b)

Reply:
Thanks @arjxn-py. Also, these environment variables have been set for the entire matrix. We need these just for the Windows steps in the job. We don't have separate Windows steps, so you can create a separate step with `Run integration tests on Windows` and add these environment variables there (similarly for unit tests, etc.)